api/graphics2_util.cpp

```diff
@@ -50,14 +50,14 @@ static void get_shmem_name(const char* prog_name, char* shmem_name) {
     APP_INIT_DATA aid;
     int retval = boinc_get_init_data(aid);
     if (retval) aid.slot = 0;
-    sprintf(shmem_name, "boinc_%s_%d", prog_name, aid.slot);
+    snprintf(shmem_name, MAXPATHLEN+1, "boinc_%s_%d", prog_name, aid.slot);
```

Comment:
This code is not safe. I'd suggest to change the function signature and add `shmem_name_size` input parameter at the end

Reply:
This change makes it more secure than it is now. Do you think changing the signature is really needed? It might break something for science apps (I haven't checked usage of `get_shmem_name`) that the projects have to deal with. The changes proposed in this PR on the other hand are sane.

Reply:
Yeah, you're probably right. Changing signature without having API versioning is a bad idea.
Also I have to admit that this API is not safe at all and should be revisited.